### PR TITLE
Fix MinGW build

### PR DESCRIPTION
**Pull request by orgads**
snprintf is defined as _snprintf, which doesn't exist in the std
namespace.

### CODE REVIEW

**Comment by orgads** [comment]
cc @tamird

**Comment by adamretter** [comment]
@orgads @tamird Just want to check that this was tested on both MinGW and Windows with Visual Studio?

**Comment by orgads** [comment]
I didn't test msvc, but the behavior is unchanged there, since `_MSC_VER` is defined.

**Comment by tamird** [comment]
MSVC should be tested in appveyor.

**Comment by adamretter** [comment]
@tamird Sadly it looks like the AppVeyor build is already broken :-( https://ci.appveyor.com/project/Facebook/rocksdb/branch/master/messages

**Comment by tamird** [comment]
Heh, indeed. I guess this PR should wait until that's fixed.

Disappointing that https://github.com/facebook/rocksdb/pull/2269 was merged with failing CI.

**Comment by yiwu-arbug** [comment]
@tamird my bad. I'll land #2287 soon.

**Comment by yiwu-arbug** [comment]
@orgads #2287 has been landed. Please rebase. Thanks!

File: port/win/port_win.h [hunk 1]
```diff
@@ -81,8 +81,8 @@ namespace rocksdb {
 
 namespace port {
 
-// VS 15
-#if (defined _MSC_VER) && (_MSC_VER >= 1900)
+// VS 15 / MinGW
+#if !defined(_MSC_VER) || (_MSC_VER >= 1900)
```
Comment (IslamAbdelRahman) [inline port/win/port_win.h]: that does not sound right to me, why will we enable this if _MSC_VER is not defined

Comment (orgads) [inline port/win/port_win.h]: Because I assume that MinGW/GCC supports C++11. Why not?

**Comment by adamretter** [comment]
@yiwu-arbug It looks like AppVeyor isn't enabled for PRs like Travis is. Could we enable it somehow?

**Comment by tamird** [comment]
@adamretter looks like it's still building them, but the status isn't being shown in GH.

https://ci.appveyor.com/project/Facebook/rocksdb/history

**Comment by adamretter** [comment]
@tamird Interesting. I think it would be good if we could get the AppVeyor "check" to show up on the PRs as well. Looks like this PR passed though, so I am happy with it +1

**Comment by yiwu-arbug** [comment]
@adamretter: @sdwilsh help fix an auth issue and the AppVeyor should be back next time.

File: port/win/port_win.h (hunk 1 above)
Comment (yiwu-arbug) [inline port/win/port_win.h]: port_win.h shouldn't enable for non-Windows. Looks like we should use `__MINGW32__` macro to tell if it is MinGW?

Comment (tamird) [inline port/win/port_win.h]: It's equivalent; this checks for MSVC. Is this change really worth more CI cycles?

Comment (tamird) [inline port/win/port_win.h]: If you wanted to be pedantic about it, you could write this as:

```
#if (defined _MSC_VER) && (_MSC_VER < 1900)
<old MSVC code>
#else
<normal code>
#endif
```

Comment (orgads) [inline port/win/port_win.h]: Moreover, I don't think there's a reason to exclude clang or ICC for example. Do they define __MINGW32__?

Comment (yiwu-arbug) [inline port/win/port_win.h]: ok, I misread the code. I thought without a mingw-specific macro the code will compile for non-windows.

btw. does MSVC has `std::snprintf` in `<cstdio>`? If so we can probably update all occurrence of `snprintf` to `std::snprintf` and get rid of the macro?

Comment (orgads) [inline port/win/port_win.h]: According to [the docs](https://msdn.microsoft.com/en-us/library/58dt9f24(v=vs.100).aspx) it was only added in MSVC2015.

Comment (tamird) [inline port/win/port_win.h]: @orgads looks like CI hasn't started on this yet (and won't for a while). Perhaps it's worth rewriting it in the style I suggested above, for clarity?

**Comment by orgads** [comment]
Rebased.

**Comment by orgads** [comment]
Rebased again.

Hello?

**Comment by orgads** [comment]
@yiwu-arbug Rebased and fixed debug and release builds. Please review.

**Comment by yiwu-arbug** [comment]
@orgads any idea why AppVeyor is failing?

**Comment by orgads** [comment]
[**@yiwu-arbug**](https://github.com/yiwu-arbug) commented on [Sep 14, 2017, 7:25 PM GMT+3](https://github.com/facebook/rocksdb/pull/2298#issuecomment-329535704 "2017-09-14T16:25:09Z - Replied by Github Reply Comments"):
> [@orgads](https://github.com/orgads) any idea why AppVeyor is failing?

Probably because of the second commit. I removed it for now, let's see if it passes and then I'll try to fix release build in a separate commit.

**Comment by orgads** [comment]
Looks good now.

**Comment by orgads** [comment]
@yiwu-arbug: Well?

**Comment by yiwu-arbug** [comment]
Sorry for delay. Merging.

**Pull request closed without merging**